libraries/adaptive-expressions/src/triggerTrees/node.ts

```diff
@@ -0,0 +1,362 @@
+/**
+ * @module adaptive-expressions
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Clause } from './clause';
+import { RelationshipType } from './relationshipType';
+import { TriggerTree } from './triggerTree';
+import { Trigger } from './trigger';
+
+enum Operation {
+    none = 'none',
+    found = 'found',
+    added = 'added',
+    removed = 'removed',
+    inserted = 'inserted'
+}
+
+/**
+ * Node in a trigger tree.
+ */
+export class Node {
+    private _allTriggers: Trigger[] = [];
+    private _triggers: Trigger[] = [];
+    private _specializations: Node[] = [];
+
+    /**
+     * Intializes a new instance of the `Node` class.
+     * @param clause The logical conjunction this node represents.
+     * @param tree The trigger tree this node is found in.
+     * @param trigger The trigger to initialize this node.
+     */
+    public constructor(clause: Clause, tree: TriggerTree, trigger?: Trigger) {
+        this.clause = new Clause(clause);
```

Comment:
Why reconstruct the `Clause`?

Reply:
I was porting the c# implementation, it is supposed to deep clone the clause here with `newClause = new Clause(fromClause)`.